mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -201,6 +213,78 @@ struct Conv2DRewritePattern : public OpRewritePattern<T> {
       }
     }
 
+    int64_t gemmM_size, gemmN_size, gemmK_size;
+    int64_t gemmM_extra, gemmN_extra, gemmK_extra;
+    gemmM_extra = gemmN_extra = gemmK_extra = 0;
+    // compute we should use extra padding kernel or not
+    if (convOpType == miopen::ConvOpType::Conv2DOpType) {
+      gemmM_size = k / g;
+      gemmK_size = (c / g) * y * x;
+      gemmN_size = n * ho * wo;
+    } else if (convOpType == miopen::ConvOpType::Conv2DBwdDataOpType) {
+      gemmM_size = c / g;
+      gemmK_size = (k / g) * y * x;
+      gemmN_size = n * ho * wo;
+    } else if (convOpType == miopen::ConvOpType::Conv2DBwdWeightOpType) {
+      gemmM_size = k / g;
+      gemmK_size = n * ho * wo;
+      gemmN_size = (c / g) * y * x;
+    }
+
+    bool needExtraPad = true;
```

Comment:
please set `needExtraPad` be `false` by default. so your new logic is used when necessarily.